clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -674,7 +673,7 @@ private KafkaConsumer(ConsumerConfig config, Deserializer<K> keyDeserializer, De
                     GroupRebalanceConfig.ProtocolType.CONSUMER);
 
             this.groupId = Optional.ofNullable(groupRebalanceConfig.groupId);
-            this.clientId = buildClientId(config.getString(CommonClientConfigs.CLIENT_ID_CONFIG), groupRebalanceConfig);
```

Comment:
Could we make the below `enableAutoCommit` check into the ConsumerConfig as well?